test/e2e/case42_resource_selection_test.go

```diff
@@ -0,0 +1,246 @@
+		By("Creating additional matching object case42-4-e2e")
+		utils.Kubectl("apply", "-n", targetNs, "-f", nomatchYaml)
+
+		Consistently(func() interface{} {
+			managedPlc := utils.GetWithTimeout(clientManagedDynamic, gvrConfigPolicy,
+				policyName, testNamespace, true, defaultTimeoutSeconds)
+
+			return utils.GetStatusMessage(managedPlc)
+		}, defaultConsistentlyDuration, 1).ShouldNot(ContainSubstring("case42-4-e2e"))
+	})
+
+	It("should evaluate when a resource is labeled to match", func() {
+		utils.Kubectl("label", "-n", targetNs, "fakeapi", "case42-4-e2e", "case42=case42-4-e2e")
+
+		Eventually(func() interface{} {
+			managedPlc := utils.GetWithTimeout(clientManagedDynamic, gvrConfigPolicy,
+				policyName, testNamespace, true, defaultTimeoutSeconds)
+
+			return utils.GetStatusMessage(managedPlc)
+		}, defaultTimeoutSeconds, 1).Should(HaveSuffix(
+			fmt.Sprintf("; fakeapis [case42-4-e2e] found but not as specified in namespace %s", targetNs)))
+	})
+
+	It("should evaluate when a matching resource label is removed", func() {
+		utils.Kubectl("label", "-n", targetNs, "fakeapi", "case42-3-e2e", "case42-")
+
+		Eventually(func() interface{} {
+			managedPlc := utils.GetWithTimeout(clientManagedDynamic, gvrConfigPolicy,
+				policyName, testNamespace, true, defaultTimeoutSeconds)
+
+			return utils.GetStatusMessage(managedPlc)
+		}, defaultTimeoutSeconds, 1).ShouldNot(ContainSubstring(
+			fmt.Sprintf("fakeapis [case42-3-e2e] found but not as specified in namespace %s", targetNs)))
+	})
+
+	It("should become compliant when enforced", func() {
+		utils.Kubectl("patch", "--namespace=managed", "configurationpolicy", policyName, "--type=json",
+			`--patch=[{
+				"op":"replace",
+				"path":"/spec/remediationAction",
+				"value":"enforce"
+			}]`,
+		)
+
+		Eventually(func() interface{} {
+			managedPlc := utils.GetWithTimeout(clientManagedDynamic, gvrConfigPolicy,
+				policyName, testNamespace, true, defaultTimeoutSeconds)
+
+			return utils.GetStatusMessage(managedPlc)
+		}, defaultTimeoutSeconds, 1).Should(Equal(fmt.Sprintf(
+			"fakeapis [case42-1-e2e] found as specified in namespace %[1]s; "+
+				"fakeapis [case42-2-e2e] found as specified in namespace %[1]s; "+
+				"fakeapis [case42-4-e2e] found as specified in namespace %[1]s", targetNs)))
+	})
+
+	It("should ignore the objectSelector when a name is provided", func() {
+		utils.Kubectl("patch", "--namespace=managed", "configurationpolicy", policyName, "--type=json",
+			`--patch=[{
+				"op": "replace",
+				"path":"/spec/object-templates/0/objectDefinition/metadata/name",
+				"value":"case42-3-e2e"
+			}]`,
+		)
+
+		Eventually(func() interface{} {
+			managedPlc := utils.GetWithTimeout(clientManagedDynamic, gvrConfigPolicy,
+				policyName, testNamespace, true, defaultTimeoutSeconds)
+
+			return utils.GetStatusMessage(managedPlc)
+		}, defaultTimeoutSeconds, 1).Should(Equal(fmt.Sprintf(
+			"fakeapis [case42-3-e2e] found as specified in namespace %s", targetNs)))
+	})
+})
```

Comment:
Do we have templated ns case?

Reply:
Good question--that was covered in:
- https://github.com/open-cluster-management-io/config-policy-controller/pull/314

And I'll probably also reuse some of those test cases for the `.ObjectName` variable.